WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV5.swift

```diff
@@ -20,4 +20,151 @@ import Foundation
+        case members
+        case messageProtocol = "protocol"
+        case messageTimer = "message_timer"
+        case mlsGroupID = "group_id"
+        case name
+        case qualifiedID = "qualified_id"
+        case readReceiptMode = "receipt_mode"
+        case teamID = "team"
+        case type
+    }
+
+    var access: [String]?
+    var accessRoles: [String]?
+    var cipherSuite: UInt16?
+    var creator: UUID?
+    var epoch: UInt?
+    var epochTimestamp: Date?
+    var id: UUID?
+    var lastEvent: String?
+    var lastEventTime: String?
+    var legacyAccessRole: String?
+    var members: QualifiedConversationMembers?
+    var messageProtocol: String?
+    var messageTimer: TimeInterval?
+    var mlsGroupID: String?
+    var name: String?
+    var qualifiedID: QualifiedID?
+    var readReceiptMode: Int?
+    var teamID: UUID?
+    var type: Int?
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        access = try container.decodeIfPresent([String].self, forKey: .access)
+        creator = try container.decodeIfPresent(UUID.self, forKey: .creator)
+        epoch = try container.decodeIfPresent(UInt.self, forKey: .epoch)
+        id = try container.decodeIfPresent(UUID.self, forKey: .id)
+        lastEvent = try container.decodeIfPresent(String.self, forKey: .lastEvent)
+        lastEventTime = try container.decodeIfPresent(String.self, forKey: .lastEventTime)
+        members = try container.decodeIfPresent(QualifiedConversationMembers.self, forKey: .members)
+        messageProtocol = try container.decodeIfPresent(String.self, forKey: .messageProtocol)
+        messageTimer = try container.decodeIfPresent(TimeInterval.self, forKey: .messageTimer)
+        mlsGroupID = try container.decodeIfPresent(String.self, forKey: .mlsGroupID)
+        name = try container.decodeIfPresent(String.self, forKey: .name)
+        qualifiedID = try container.decodeIfPresent(QualifiedID.self, forKey: .qualifiedID)
+        readReceiptMode = try container.decodeIfPresent(Int.self, forKey: .readReceiptMode)
+        teamID = try container.decodeIfPresent(UUID.self, forKey: .teamID)
+        type = try container.decodeIfPresent(Int.self, forKey: .type)
+
+        // v3 replaces the field "access_role_v2" with "access_role".
+        // However, since the format of update events does not depend on versioning,
+        // we may receive conversations from the `conversation.create` update event
+        // which still have both "access_role_v2" and "access_role" fields
+
+        if container.contains(CodingKeys.accessRoleV2) {
+            legacyAccessRole = try container.decodeIfPresent(String.self, forKey: .accessRole)
+            accessRoles = try container.decodeIfPresent([String].self, forKey: .accessRoleV2)
+        } else {
+            legacyAccessRole = nil
+            accessRoles = try container.decodeIfPresent([String].self, forKey: .accessRole)
+        }
```

Comment:
Do I understand that this logic is necessary in order to use this `ConversationsAPIV5` for both payloads from /conversations as well as payloads in update events?